src/components/organism/CreatePool/index.tsx

```diff
@@ -0,0 +1,338 @@
+    .toLocaleString()
+    ?.replace(/[, ]/g, "");
+
+  const navigateToPools = () => {
+    navigate(POOLS_PAGE);
+  };
+
+  const checkIfPoolAlreadyExists = (id: string) => {
+    let exists = false;
+
+    if (id) {
+      exists = !!pools.find((pool: any) => {
+        return (
+          pool?.[0]?.[1]?.interior?.X2 &&
+          pool?.[0]?.[1]?.interior?.X2?.[1]?.GeneralIndex?.replace(/[, ]/g, "").toString() === id
+        );
+      });
+    }
+
+    setPoolExists(exists);
+  };
+
+  const handlePool = async () => {
+    try {
+      if (api) {
+        await createPool(
+          api,
+          selectedTokenB.assetTokenId,
+          selectedAccount,
+          nativeTokenValue,
+          assetTokenValue,
+          nativeTokenWithSlippage.tokenValue.toString(),
+          assetTokenWithSlippage.tokenValue.toString(),
+          dispatch
+        );
+      }
+    } catch (error) {
+      dotAcpToast.error(`Error: ${error}`);
+    }
+  };
+
+  const handlePoolGasFee = async () => {
+    if (api) await checkCreatePoolGasFee(api, selectedTokenB.assetTokenId, selectedAccount, dispatch);
+  };
+
+  const closeSuccessModal = async () => {
+    setIsSuccessModalOpen(false);
+    dispatch({ type: ActionType.SET_SUCCESS_MODAL_OPEN, payload: false });
+    if (api) await getAllPools(api);
+    navigateToPools();
+  };
+
+  const setSelectedTokenAValue = (value: number) => {
+    if (selectedTokenNativeValue && slippageValue) {
+      const nativeTokenSlippageValue = calculateSlippageReduce(value, slippageValue);
+      const tokenWithSlippageFormatted = formatInputTokenValue(nativeTokenSlippageValue, selectedTokenB?.decimals);
+      setSelectedTokenNativeValue({ tokenValue: value });
+      setNativeTokenWithSlippage({ tokenValue: parseInt(tokenWithSlippageFormatted) });
+    }
+  };
+
+  const setSelectedTokenBValue = (value: number) => {
+    if (selectedTokenAssetValue && slippageValue) {
+      const assetTokenSlippageValue = calculateSlippageReduce(value, slippageValue);
+      const tokenWithSlippageFormatted = formatInputTokenValue(assetTokenSlippageValue, selectedTokenB?.decimals);
+      setSelectedTokenAssetValue({ tokenValue: value });
+      setAssetTokenWithSlippage({ tokenValue: parseInt(tokenWithSlippageFormatted) });
+    }
+  };
+
+  const checkAssetTokenMinAmount = async () => {
+    if (selectedTokenAssetValue && api && selectedTokenB.assetTokenId) {
+      const assetTokenInfo: any = await api.query.assets.asset(selectedTokenB.assetTokenId);
+      const assetTokenMinBalance = assetTokenInfo.toHuman()?.minBalance;
+      if (selectedTokenAssetValue.tokenValue >= assetTokenMinBalance?.replace(/[, ]/g, "")) {
+        setAssetTokenMinValueExceeded(false);
+      } else {
+        setAssetTokenMinValue(assetTokenMinBalance);
+        setAssetTokenMinValueExceeded(true);
+      }
+    }
+  };
+
+  const returnButtonStatus = () => {
```

Comment:
This could be wrapped with useMemo. We do not need to rerender all the time e.g.: if u open a modal. We need this logic only if the selected token is changed.